### PR TITLE
Add yml validation schema generation and schema checker

### PR DESCRIPTION
**Pull request by ClemensLinnhoff**
#### Reference to a related issue in the repository
#64 

#### Add a description
- Automatically generate yml schema files by rules2yml.py. The schema files are generated in <rules_folder>/schema.
- Validate each rule file with the respective schema before loading it.

#### Mention a member
@masipp, @jdsika 

#### Check the checklist

- [x] My code and comments follow the [contributors guidelines](https://opensimulationinterface.github.io/osi-documentation/osi/howtocontribute.html) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the [documentation](https://github.com/OpenSimulationInterface/osi-documentation) for osi-validation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests / travis ci pass locally with my changes.

### CODE REVIEW

**Comment by jdsika** [comment]
@ClemensLinnhoff we have some conflicts now

**Comment by ClemensLinnhoff** [comment]
> @ClemensLinnhoff we have some conflicts now

I fixed the conflicts. Ready for review.

File: doc/usage.adoc [hunk 1]
```diff
@@ -59,6 +59,11 @@ python rules2yml.py # Generates the rule directory
 osivalidator --data data/20240221T141700Z_sv_300_2112_10_one_moving_object.osi --rules rules -p
 ----
 
+The rules2yml.py generates yml files for all OSI proto files containing the rules specified in OSI.
+It also generated respective yml schema files to validate the rule yml files.
+The schema files are located in <rules_folder>/schema.
+When executing osivalidator, all rule files are validated against their respective schema.
```
Comment (jdsika) [inline doc/usage.adoc:65]: It parses the rules in combination with their application in the original osi files. Why would the schema be coupled to the filename? Isn't the base idea that someone creates custom yml files (setting a speedlimit on an object) and edits existing ones and wants a checker for those?

Comment (ClemensLinnhoff) [reply doc/usage.adoc:65]: The schema files do not contain rules for specific fields. All rules are allowed for all fields. So having custom rules files is supported.

Comment (ClemensLinnhoff) [reply doc/usage.adoc:67]: I adjusted the documentation.

**Review by jdsika — APPROVED** [review]

**Pull request merged**